### ISSUE TITLE
Secret Manager cannot deserialize plain text based secrets

### ISSUE THREAD

**Issue by anthofo** [awspring/spring-cloud-aws#337]
**Type**: Bug

**Component**:
Secrets Manager

**Describe the bug**
On AWS Secret Manager it is possible to define a secret as a list of key/value (json) or a plain text (string):
![Capture d’écran 2022-04-21 181457](https://user-images.githubusercontent.com/3809685/164505560-3ec95e58-cd2b-4237-b7cb-5f9143e0e22d.png)
The plain text can be used to store rsa keys, for instance.
But the implementation in `AwsSecretsManagerPropertySource` doesn't allow plain text secrets, only json based secrets. When trying to use a plain text we end up with a json parsing error:

```	at io.awspring.cloud.secretsmanager.AwsSecretsManagerPropertySource.readSecretValue(AwsSecretsManagerPropertySource.java:91)
	at io.awspring.cloud.secretsmanager.AwsSecretsManagerPropertySource.init(AwsSecretsManagerPropertySource.java:64)
	at io.awspring.cloud.secretsmanager.AwsSecretsManagerPropertySources.createPropertySource(AwsSecretsManagerPropertySources.java:87)
	... 10 common frames omitted
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value:
```

**Solution**
Change `AwsSecretsManagerPropertySource` to handle plain text secrets if a json is not detected by adding a `catch (JsonParseException e)` block:

```java
GetSecretValueResult secretValueResult = source
		.getSecretValue(secretValueRequest);
try {
	Map<String, Object> secretMap = jsonMapper.readValue(
		secretValueResult.getSecretString(),
		new TypeReference<Map<String, Object>>() {
		});

	for (Map.Entry<String, Object> secretEntry : secretMap.entrySet()) {
		properties.put(secretEntry.getKey(), secretEntry.getValue());
	}
} catch (JsonParseException e) {
	// If the secret is not a JSON string, then it is a simple string
	properties.put(secretValueResult.getName(), secretValueResult.getSecretString());
}
````

**Comment by maciejwalkowiak** [comment]
Thanks @anthofo for reporting and preparing a PR. I'll take a deeper look next week.

**Comment by maciejwalkowiak** [comment]
Fixed in #338
